crates/compiler/gen_dev/src/generic64/mod.rs

```diff
@@ -1758,6 +1758,93 @@ impl<
         self.free_symbol(&Symbol::DEV_TMP3);
     }
 
+    fn build_list_prepend(
+        &mut self,
+        dst: &Symbol,
+        args: &'a [Symbol],
+        arg_layouts: &[InLayout<'a>],
+        ret_layout: &InLayout<'a>,
+    ) {
+        let list = args[0];
+        let list_layout = arg_layouts[0];
+        let elem = args[1];
+        let elem_layout = arg_layouts[1];
+
+        // List alignment argument (u32).
+        let u32_layout = Layout::U32;
+        let list_alignment = self.layout_interner.alignment_bytes(*ret_layout);
+        self.load_literal(
+            &Symbol::DEV_TMP,
+            &u32_layout,
+            &Literal::Int((list_alignment as i128).to_ne_bytes()),
+        );
+
+        // Have to pass the input element by pointer, so put it on the stack and load it's address.
+        self.storage_manager
+            .ensure_symbol_on_stack(&mut self.buf, &elem);
+        let (new_elem_offset, _) = self.storage_manager.stack_offset_and_size(&elem);
+
+        // Load address of input element into register.
+        let reg = self
+            .storage_manager
+            .claim_general_reg(&mut self.buf, &Symbol::DEV_TMP2);
+        ASM::add_reg64_reg64_imm32(&mut self.buf, reg, CC::BASE_PTR_REG, new_elem_offset);
+
+        // Load element_witdh argument (usize).
+        let u64_layout = Layout::U64;
+        let elem_stack_size = self.layout_interner.stack_size(elem_layout);
+        self.load_literal(
+            &Symbol::DEV_TMP3,
+            &u64_layout,
+            &Literal::Int((elem_stack_size as i128).to_ne_bytes()),
+        );
```

Comment:
I think we use this pattern enough to justify a method that, given an `InLayout` (`elem_layout` here) and a symbol (the `Symbol::DEV_TMP3` here), loads the layout width. e.g. 

```rust
    fn load_layout_stack_size(&mut self, layout: InLayout<'a>, symbol: Symbol) {
        let u64_layout = Layout::U64;
        let width = self.layout_interner.stack_size(layout);
        let width_literal = Literal::Int((width as i128).to_ne_bytes());

        self.load_literal(&symbol, &u64_layout, &width_literal);
    }
``` 

this helper can go in the impl block that is around line 2300.

Reply:
Sounds good, i applied this in 40c742d74d518e46e1f62c604cf1a15de3921cdd

I also extracted an equivalent pattern in 74c64bf431640edc43ce2edccb179c042d3157ba for layout alignment. I'll make sure to try to keep this in mind with upcoming things I work on :)